code/modules/mob/living/simple_animal/bot/vibebot.dm

```diff
@@ -55,3 +58,51 @@
 		start_patrol()
 	if(mode == BOT_PATROL)
 		bot_patrol()
+
+/mob/living/simple_animal/bot/vibebot/turn_off()
+	remove_atom_colour(TEMPORARY_COLOUR_PRIORITY)
+	. = ..()
+
+/mob/living/simple_animal/bot/vibebot/proc/vibe()
+	remove_atom_colour(TEMPORARY_COLOUR_PRIORITY)
+	add_atom_colour("#[random_color()]", TEMPORARY_COLOUR_PRIORITY)
+	set_light_color(color)
+
+/**
+ * Vibebot's vibe ability
+ *
+ * Given to sentient vibebots so they can also vibe.
+ */
+/datum/action/innate/vibebot_vibe
+	name = "Vibe"
+	desc = "Change your vibebot color."
+	icon_icon = 'icons/mob/aibots.dmi'
+	button_icon_state = "vibebot"
+	///The vibebot this action is stored to
+	var/mob/living/simple_animal/bot/vibebot/bot_mob
```

Comment:
Don't do this, use the `target` var

When you create the action, set it's target in `new()` to the vibe bot
Then you can validate if the target is the correct mob type